aten/src/ATen/native/native_functions.yaml

```diff
@@ -6642,6 +6642,12 @@
   variants: function, method
   device_guard: False
 
+- func: _add_batch_dim(Tensor self, int batch_dim, int level) -> Tensor
+  use_c10_dispatcher: full
```

Comment:
Make these function only?

Reply:
Is the intention to eventually delete these functions when we have more vmap functionality? If not, why do we need these functions--does it make sense to make these functions part of the *public* API for batched tensors?

Reply:
iirc the default for native_functions.yaml is function only. I can the `variants: function` line to make that explicit, though

Reply:
> Is the intention to eventually delete these functions when we have more vmap functionality? If not, why do we need these functions--does it make sense to make these functions part of the public API for batched tensors?

These functions are part of the pytorch developer API for batched tensors. However, pytorch users shouldn't be exposed to batched tensors, so they should never see these functions.

I was not planning on deleting these functions, but I think it makes sense to put them in a namespace in the future (so they do not appear as functions on torch)

Reply:
Probably would be a good idea to documented the specific situations where these functions are intended to be used, so it's easier for us to assess if other future users are using them inappropriately.

Reply:
Documented.